doc/experimental_ui_api.md

```diff
@@ -0,0 +1,2027 @@
+purposes by placing related styles in an empty parent style. Omitting the
+selector causes it to be automatically set to the universal selector `*`. For
+example, this style might be used for the window's `style` property:
+
+```lua
+ui.Style {
+    ui.Style "button, toggle, option" {
+        -- Style properties for all the different types of buttons.
+    },
+    ui.Style "check, switch" {
+        -- Style properties for checkboxes and switches.
+    },
+    ui.Style "radio" {
+        -- Style properties for radio buttons.
+    },
+}
+```
+
+### Style resets
+
+Sometimes, it is desireable to remove all existing style from a certain element
+type. This can be done via _style resets_. The `reset` boolean, when set,
+resets all style properties for the selected elements to their default values.
+
+For instance, if one style gives buttons a red background and nonzero padding,
+then setting the `reset` property on a later style for buttons will reset that
+to the defaults of a transparent background and no padding. Note that this will
+also reset any properties set by the prelude theme, described in [Theming].
+
+```lua
+-- One style somewhere adds a bunch of properties to buttons.
+ui.Style "button" {
+    box_fill = "red",
+    padding = {2, 2, 2, 2},
+}
+
+-- The button with the ID "special" needs unique styling, and hence uses the
+-- `reset` property to get a clean slate for styling.
+ui.Style "#special" {
+    reset = true,
+    text_color = "yellow",
+}
+```
+
+Style resets cascade with style states as well. Resetting buttons with the
+`hovered` state will also reset the properties for buttons that are both
+hovered and pressed, for instance. Resetting `button` directly will reset every
+state for the button as well.
+
+Theming
+-------
+
+It is usually the case that different windows share the same basic element
+styles. This concept is supported natively by the UI API through the use of
+_themes_, which are large groups of styles that can be used in the `theme`
+property of windows or set as the globally default theme.
+
+If a window doesn't have the `theme` property set, it will automatically use
+the _default theme_, which is retrieved by `ui.get_default_theme()`. The engine
+initially sets the default theme to the prelude theme, but games and mods can
+change the default theme with the `ui.set_default_theme()` function. In the
+future, the UI API will include a default theme called the "core theme", which
+games can optionally use if they don't want to make their own theme or are in
+early stages of development.
+
+### Prelude theme
+
+Nearly all themes should be based on the _prelude theme_, a theme that serves
+no other purpose than being a base for other themes. The prelude theme sets
+style properties that are essential to the functionality of the element, or are
+at least a basic part of the element's intended design. It can be accessed with
+the `ui.get_prelude_theme()` function.
```

Comment:
Core theme vs prelude theme, will they be one in the same or no?

Reply:
They are separate. The prelude is designed to be the base for every theme (including the core theme), and doesn't give any visuals. Imagine the prelude theme as a web browser's default stylesheet after a CSS reset has been used--things like `<head>` tags are still hidden, but other things like buttons don't have a distinct look from any other element.